### PR TITLE
API to create an image file

### PR DESCRIPTION
**Pull request by Dmitry-Borodin**
Changed "image" crate to "png" crate dependency, now instead of 1.5-2Mb, we have ~200kb dependency according to lib.rs
Added this dependency for cairo-backend, actually can be implemented for direct2d as well (copy&paste the code?)
#128

### CODE REVIEW

File: piet-common/src/cairo_back.rs [hunk 1]
```diff
@@ -121,4 +126,19 @@ impl<'a> BitmapTarget<'a> {
         }
         Ok(raw_data)
     }
+
```
Comment (cmyr) [inline piet-common/src/cairo_back.rs]: We try to have documentation for all public items

File: piet-common/src/cairo_back.rs [hunk 2]
```diff
@@ -121,4 +126,19 @@ impl<'a> BitmapTarget<'a> {
...
+    pub fn save_to_file(self, path: &str) -> Result<(), piet::Error> {
```
Comment (cmyr) [inline piet-common/src/cairo_back.rs]: for `path`, I think the best type for the signature would be `impl AsRef<Path>`; this allows a larger number of acceptable types to be used here, and is the same type that `File::create` takes.

File: piet-common/src/cairo_back.rs [hunk 3]
```diff
@@ -121,4 +126,19 @@ impl<'a> BitmapTarget<'a> {
...
+        let height = self.surface.get_height();
+        let width = self.surface.get_width();
+        let image = self.into_raw_pixels(ImageFormat::RgbaPremul)?;
+        let file = BufWriter::new(File::create(path)
+            .map_err(|e| Into::<Box<dyn std::error::Error>>::into(e))?);
```
Comment (cmyr) [inline piet-common/src/cairo_back.rs]: I was playing around trying to figure out if there was a less verbose way to do this conversion, and the best I can come up with is,

```rust
.map_err(|e| Box::new(e) as Box<_>)?;
// also works:
.map_err(|e| Into::<Box<_>>::into(e))?;
```
In any case this is fine, I was just playing around.

Comment (Dmitry-Borodin) [reply piet-common/src/cairo_back.rs]: That truly looks better

**Comment by Dmitry-Borodin** [comment]
Thank you for review. FIxed comments, implemented the same functionality for d2d.
And I don't understand why CI is failing.

**Comment by cmyr** [comment]
might just be `rustfmt` failing?

**Comment by Dmitry-Borodin** [comment]
I see a problem, need to use different implementation for windows, will fix it soon.

**Comment by cmyr** [comment]
cool, this looks okay; I'm not going to be able to test out the windows version for a couple days, but  it looks good!

**Comment by cmyr** [comment]
@Dmitry-Borodin if you're interested in a project that takes advantage of this patch, I've written up some thoughts in https://github.com/xi-editor/druid/issues/527. If you end up experimenting with this, and run into a lifetime issue with the `Text` struct, change the `'static` to `'a`. (sorry if this is cryptic, there's just another recent breaking change in piet that you might encounter)

**Comment by Dmitry-Borodin** [comment]
Thank you for information, will play around with it.

**Comment by cmyr** [comment]
Okay, this is working great for me. Would like to get a +1 from @raphlinus to make sure the dependency change is okay, but I'm very happy.

**Comment by cmyr** [comment]
Okay, one final request here which is that we make `png` a feature. This will mean we will also need to put this new method behind a feature flag. A nice thing we could do, here, is something like:

```rust
/// Save bitmap to RGBA PNG file
#[cfg(feature = "svg")]
pub fn save_to_file<P: AsRef<Path>>(self, path: P) -> Result<(), piet::Error> { ... }

#[cfg(not(feature = "svg"))]
pub fn save_to_file<P: AsRef<Path>>(self, path: P) -> Result<(), piet::Error> { 
    Err(piet::error::new_error(piet::error::ErrorKind::MissingFeature))
}
```

That is, the api would still exist if the feature was disabled, just we would return an error that would communicate the problem.

**Comment by hwchen** [comment]
I've just added github actions to the CI in master (#138), I'm not exactly sure how it will affect this PR.

The new CI has clippy lints turned on, so I think you'll have to rebase. (I think I've caught and cleaned up some of those lints since this branch was started).

Let me know if you have any questions.

**Comment by Dmitry-Borodin** [comment]
Will continue working on it to avoid warnings and clarify how features are working

**Comment by Dmitry-Borodin** [comment]
Sample doesn't work properly on my machine, investigating. Like feature is always missing.

**Comment by Dmitry-Borodin** [comment]
@cmyr implemented and tested on GNU/Linux. It's done now.
CI have some package download problems.

**Comment by hwchen** [comment]
Re: CI. When on ubuntu, `sudo apt update` has the error:
```
Err:28 https://packages.microsoft.com/repos/microsoft-ubuntu-bionic-prod bionic/main amd64 Packages
  File has unexpected size (94576 != 93944). Mirror sync in progress? [IP: 40.76.35.62 443]
  Hashes of expected file:
   - Filesize:93944 [weak]
```

@Dmitry-Borodin was the CI working before? Judging from the commit history in this PR, it appears it was working before the last commit.

Perhaps this will just fix itself if we wait...

**Comment by Dmitry-Borodin** [comment]
Yes, it was working before and it's ok now. Somebody restarted checks?

**Pull request merged**